strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassDependenciesGraph.java

```diff
@@ -217,24 +269,41 @@ private boolean addImport(JavaElement javaElementParentClass, String importz) {
         return false;
     }
 
+    Set<String> findTestsDependingOn(Collection<Change> files) {
 
-    Set<String> findTestsDependingOn(Set<File> classes) {
-        return classes.stream()
-            .map( javaClass -> {
-                final File classLocation = JavaToClassLocation.transform(javaClass, testVerifier);
-                final String className = this.builder.getClassName(classLocation);
-                return this.builder.getClassDescription(className);
-            })
-            .map(JavaElement::new)
+        return files.stream()
+            .map(Change::getLocation)
+            .map(this::transformToElement)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
             .filter(graph::containsVertex)
             .map(this::getParents)
             .flatMap(Collection::stream)
+            .map(e -> (JavaElement) e)
             .map(JavaElement::getClassName)
             .collect(Collectors.toSet());
```

Comment:
Can't we somehow combine some operations to one? I am not sure what is the impact to performance here (I would assume consecutive `map`s are combined for example), but I'm more thinking about the readability here. We flip between domain objects (where domain is a `Change` and `JavaElement`) and `Optional`s for example.